mvc/views/dependency-injection.rst

```diff
@@ -1,93 +1,79 @@
+
+.. TODO: Add localization sample in RC2 timeframe.
+
+`View sample files <https://github.com/aspnet/Docs/tree/1.0.0-rc1/mvc/views/dependency-injection/sample>`_
+
+.. warning:: When applying this functionality, ensure you maintain `separation of concerns <http://deviq.com/separation-of-concerns>`_ between your controllers and views. Most of the data your views  display should be passed in from the controller. This technique is recommended only for view-specific data.
+
+A Simple Example
+----------------
+With the above warning in mind, you can inject a service into a view using the ``@inject`` directive, as shown below:
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Views/Todo/Index.cshtml
   :linenos:
+  :language: c#
+  :emphasize-lines: 4-5,15-17
+
+This view displays a list of ``ToDoItem`` instances, along with a summary showing overall statistics. The summary is populated from the injected ``StatisticsService``. This service is registered for dependency injection in ``ConfigureServices`` in ``Startup.cs``:
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Startup.cs
+  :linenos:
+  :lines: 15-22
+  :language: c#
+  :emphasize-lines: 5-6
+  :dedent: 8
   
-  @inject TodoList.Services.StatisticsService Statistics
-
-3. Add markup calling the StatisticsService to the end of the file:
-
-  .. code-block:: html
-    :linenos:
-    :emphasize-lines: 6-11
-    
-    @* Markup removed for brevity *@
-    <div>@Html.ActionLink("Create New Todo", "Create", "Todo") </div>
-    </div>
-      <div class="col-md-4">
-        @await Component.InvokeAsync("PriorityList", 4, true)
-        <h3>Stats</h3>
-        <ul>
-          <li>Items: @await Statistics.GetCount()</li>
-          <li>Completed:@await Statistics.GetCompletedCount()</li>
-          <li>Average Priority:@await Statistics.GetAveragePriority()</li>
-        </ul>
-      </div>
-    </div>
-
-4. Register the ``StatisticsService`` class in the *Startup.cs* file: 
-
-  .. code-block:: c#
-    :linenos:
-    :emphasize-lines: 8
-    
-    public void ConfigureServices(IServiceCollection services)
-    {
-      // Code removed for brevity.
-      // Add MVC services to the services container.
-      services.AddMvc();
-      services.AddTransient<TodoList.Services.StatisticsService>();
-    }
-
-The statistics are displayed:
- 
-.. image:: dependency-injection/_static/stat.png
+The ``StatisticsService`` simply performs some calculations on the set of ``ToDoItem`` instances, which it accesses via a repository:
+
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Model/Services/StatisticsService.cs
+  :linenos:
+  :language: c#
+  :emphasize-lines: 16,21,27
+
+.. note:: In this sample the repository uses an in-memory collection. The implementation shown above is not recommended for large, remotely accessed data sets.
```

Comment:
Updated; I don't want to say in-memory collections are not recommended for production (they should be fine). My point was mainly to call out that my filtering logic is all being done in-memory, which would perform poorly if I were running this code against a DbContext and remote database.